File: src/main/java/org/hisp/dhis/rules/models/RuleEffect.java
package org.hisp.dhis.rules.models;

import javax.annotation.CheckForNull;
import javax.annotation.Nonnull;

;

public record RuleEffect( @Nonnull String ruleId, @Nonnull RuleAction ruleAction, @CheckForNull String data )
{
    public RuleEffect( @Nonnull String ruleId, @Nonnull RuleAction ruleAction )
    {
        this( ruleId, ruleAction, "" );
    }
}

File: src/main/java/org/hisp/dhis/rules/models/RuleEffect.kt
package org.hisp.dhis.rules.models

import javax.annotation.Nonnull

data class RuleEffect(
    val ruleId: String,
    val ruleAction: RuleAction,
    val data: String = ""
)
{
    fun data() : String {
        return data
    }

    fun ruleAction() : RuleAction {
        return ruleAction
    }
}